### ISSUE TITLE
bash: cannot set terminal process group (-1): Inappropriate ioctl for device

### ISSUE THREAD

**Issue by lifeng68** [lxc/lxc#1804]
When i create a container, and set the `lxc.init.cmd = /bin/bash` as follow:

```
lifeng/github/lxc ‹Modify_console› » cat /usr/local/var/lib/lxc/my-container1/config
# Template used to create this container: /usr/local/share/lxc/templates/lxc-download
# Parameters passed to the template:
# For additional config options, please look at lxc.container.conf(5)

# Uncomment the following line to support nesting containers:
#lxc.include = /usr/local/share/lxc/config/nesting.conf
# (Be aware this has security implications)


# Distribution configuration
lxc.include = /usr/local/share/lxc/config/ubuntu.common.conf
lxc.arch = x86_64

# Container specific configuration
lxc.rootfs.path = dir:/usr/local/var/lib/lxc/my-container1/rootfs
lxc.uts.name = my-container1
lxc.init.cmd = /bin/bash
```
And i started the container with lxc-start, the output print` bash: cannot set terminal process group (-1): Inappropriate ioctl for device `. I tried to debug that,but i didn't solve the problem. Can you give me some advice on that ?

```
lifeng/github/lxc ‹Modify_console› » lxc-start -n my-container1 -F                 
bash: cannot set terminal process group (-1): Inappropriate ioctl for device
bash: no job control in this shell
root@my-container1:/# 
root@my-container1:/# 
root@my-container1:/# 
```

**Comment by brauner** [comment]
Right, so this is an interesting problem that I actually just discussed today. The error stems from a `glibc` patch @hallyn and I wrote a while back and pushed to upstream. Prior to our patch `ttyname{_r}()` would report `NULL` to the caller when issued on a system with an unpatched `glibc`. The reason for that was that `ttyname{_r}()` could not deal with `{p,t}ty` devices from `devpts` mounts in different mount namespaces and so tools like `tty` reported back to the caller `no tty` when in fact it was. So the `bash` shell detected this and didn't try to setup job control in these cases which is why you didn't see these errors before. No, a `glibc` with our patch will handle the `/dev/console` case perfectly fine and report back that it is in fact attached to a `{p,t}ty`. But now `bash` will try to initialize job control but will because of mount namespaces (I'm glossing over details now.) will not be able to do this correctly. So I believe this to be a bug in `bash` that should handle this case differently. In any case, short of mounting a `{p,t}ty` fd from the container's devpts mount in it's own mount namespace there's nothing `liblxc` can really do here.
